src/parsers/manifestjson.js

```diff
@@ -223,6 +223,12 @@ export default class ManifestJSONParser extends JSONParser {
     }
 
     if (this.parsedJSON.dictionaries) {
+      const {applications: {gecko: {
+        id: addonId = null} = {}} = {}} = this.parsedJSON;
```

Comment:
This took quite some time to understand tbh and I still don't quite get it :man_shrugging:

we already have two other places that traverse their way through `parsedJson.applications.gecko` so at this point I think it may make sense to do something like

```javascript
const geckoData = (this.parsedJSON.applications || {}).gecko || {}
```

to have some default values and re-use that in https://github.com/mozilla/addons-linter/pull/2195/files#diff-baa432d9354e3de899d28a2638d1499eR257 and following lines.

What do you think?

Reply:
Sure, that works for me. I can also use a `try`/`catch` if you prefer.

Reply:
Eh actually we have a `getAddonId()` function that does it with a `try`/`catch`. I'll use that.